src/main/java/org/alliancegenome/curation_api/model/entities/bulkloads/BulkLoadFile.java

```diff
@@ -56,6 +59,6 @@ public String getS3Url() {
     @JsonIgnore
     @JsonView({View.FieldsOnly.class})
     public String generateS3MD5Path() {
-        return md5Sum.charAt(0) + "/" + md5Sum.charAt(1) + "/" + md5Sum.charAt(2) + "/" + md5Sum.charAt(3) + "/" + md5Sum + ".gz";
+        return md5Sum.charAt(0) + "/" + md5Sum.charAt(1) + "/" + md5Sum.charAt(2) + "/" + md5Sum.charAt(3) + "/" + md5Sum + "." + fileType + ".gz";
```

Comment:
Actually upon more investigation I think the only change that needs to happen is  `md5Sum + "." + fileType + ".gz"` -> `md5Sum + "." + bulkLoad.getFileExtension() + ".gz"` and the fileExtension field needs to be added to the BulkLoad object only and that will take care of everything. And the UI needs to set that field upon creation.